smart_campus/app/models.py

```diff
@@ -1,3 +1,222 @@
 from django.contrib.gis.db import models
+from django.contrib.auth.base_user import (
+    BaseUserManager, AbstractBaseUser
+)
 
-# Create your models here.
+class Beacon(models.Model):
+    beacon_id = models.CharField(max_length=200, primary_key=True)
+    name = models.CharField(max_length=200)
+    description = models.CharField(max_length=200, blank=True)
+    location = models.GeometryField(srid=4326, null=True)
+    owner_group = models.ForeignKey('UserGroup', null=True, on_delete=models.SET_NULL)
+    station = models.ForeignKey('Station', null=True, on_delete=models.SET_NULL)
+
+    def __str__(self):
+        return self.name
+
+class UserManager(BaseUserManager):
+    def create_user(self, email, password=None):
+        """
+        Creates and saves a User with the given email and password
```

Comment:
"Create and save a User instance with the given email and password" might be better